### PR TITLE
Basic UI: fix live update (Microsoft Edge)

### LINKED ISSUE #5028: Basic UI with Microsoft Edge: page content hidden after an update event is received
Testing OH snapshot 1203, I discovered that each time an event update is received, the page content is make unvisible in Basiuc UI when using Microsoft Edge. F5 allows restoring the page.
It is certainly a consequence of my change in PR #4857.
Strange, it was working in my Eclipse dev environment.
I will provide a fix very soon as it is a major issue for any user using Microsoft Edge as browser.

### LINKED ISSUE #5029: Basic UI with Microsoft Edge: widgets not updating in Group Demo page
Using the demo sitemap with Basic UI and Microsoft Edge, I noticed that the page "Group Demo" is not refreshed.

### PR DESCRIPTION
**Pull request by lolodomo**
Fixes #5028
Fixes #5029

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@resetnow @kaikreuzer : a quick review and merge would be appreciated as the current live update using Microsoft Edge is currentlmy "broken".

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js [hunk 1]
```diff
@@ -1589,7 +1591,9 @@
 			smarthome.dataModel = {};
 
 			function appendControl(control) {
-				smarthome.dataModel[control.id] = control;
+				if (control.id.length > 0) {
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: Any reason to perform this check? Can widget ID ever be empty?

I can test it later today, as for now I only have this question.

Comment (lolodomo) [reply extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: Yes it happens when a frame is created when not defined in the sitemap, like for example in the "Astronomical Data" or 'Group Demo' pages in the demo sitemap.

**Comment by resetnow** [comment]
Works for me.

Kind of my fault — I noticed this in a previous PR, but wasn't sure if it's due to my environment setup or it's actually a problem in the code, and since it takes forever to setup everything from ground up I didn't test it properly.

**Pull request merged**